src/Microsoft.ML.AutoML/AutoMLExperiment/TrialResult.cs

```diff
@@ -14,4 +20,20 @@ public class TrialResult
 
         public double DurationInMilliseconds { get; set; }
     }
+
+    /// <summary>
+    /// TrialResult with Binary Classification Metrics
+    /// </summary>
+    internal class BinaryClassificationTrialResult : TrialResult
```

Comment:
Do we not want the user to have access to this class?

Reply:
Nope, we just want user to have access to `TrialResult` only. The difference between `BinaryClassificationTrialResult` and `TrialResult` is the first one contains all binary classification metrics, which is for backward compatibility and avoid changing the behavoir of old AutoML only. So I'd rather `BinaryClassificationTrialResult` not expose to public

Reply:
Will they not need the extra information? If not then sounds good.

Reply:
No they shouldn't need those extra metric informations. I don't understand why the old AutoML.Net API returns all metrics together even though the tuner only optimizes based on one of them. It's probably just a mis-design.